### PR TITLE
removed warnings of clang@11 compiler

### PR DESCRIPTION
**Pull request by ravil-mobile**
* added submodules/async via system system
* added -Wno-unused-parameter flag for clang

### CODE REVIEW

**Comment by ravil-mobile** [comment]
@krenzland , can you have a look at ?
```
ODEVector.h:86:14: warning: definition of implicit copy constructor for 'ODEVector' is deprecated because it has a user-provided copy assignment operator
```

File: src/Initializer/MemoryAllocator.cpp [hunk 1]
```diff
@@ -79,7 +79,7 @@
 
 void* seissol::memory::allocate(size_t i_size, size_t i_alignment, enum Memkind i_memkind)
 {
-    void* l_ptrBuffer;
+    void* l_ptrBuffer{};
```
Comment (sebwolf-de) [inline src/Initializer/MemoryAllocator.cpp:82]: Maybe `=nullptr` would be more verbose?

Comment (krenzland) [reply src/Initializer/MemoryAllocator.cpp:82]: Imo that is clear enough, nulptr is obviously the default value

**Comment by ravil-mobile** [comment]
Hi, I decided to remove all `NULL` from `seissol::memory::allocate` replace them with `nullptr`.

**Comment by krenzland** [comment]
> @krenzland , can you have a look at ?
> 
> ```
> ODEVector.h:86:14: warning: definition of implicit copy constructor for 'ODEVector' is deprecated because it has a user-provided copy assignment operator
> ```

Hm, it probably means that we need to define a copy constructor manually for this class?

**Comment by ravil-mobile** [comment]
Removed `ODEVector.h:86:14: warning: definition of implicit copy constructor for 'ODEVector' is deprecated because it has a user-provided copy assignment operator`

@krenzland, I just added
```
ODEVector(const ODEVector&) = default;
```
because all methods attributes have their defined copy-constructors

**Review by sebwolf-de — APPROVED** [review]

**Pull request merged**